.github/workflows/test.yml

```
@@ -0,0 +1,48 @@
+name: Test
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      max-parallel: 5
+      matrix:
+        python-version: ['3.6', '3.7', '3.8', '3.9', 'pypy3']
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Get pip cache dir
+      id: pip-cache
+      run: |
+        echo "::set-output name=dir::$(pip cache dir)"
+
+    - name: Cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.pip-cache.outputs.dir }}
+        key:
+          ${{ matrix.python-version }}-v1-${{ hashFiles('**/setup.py') }}
```

We also have deps in `tox.ini`, we can include it here:
```suggestion
          ${{ matrix.python-version }}-v1-${{ hashFiles('**/setup.py') }}-${{ hashFiles('**/tox.ini') }}
```

Reasoning here: https://github.com/jazzband/pip-tools/pull/1245#discussion_r531204310

---

Ah wait, are tox deps installed into a virtualenv and then do they skip usual pip cache?

---

I think tox just uses [pip's local cache for a while now](https://github.com/tox-dev/tox/blob/7290a2ddad3728b94bfccdd358d028e9ba3ca5a1/src/tox/config/__init__.py#L704) unless the [default `install_command`](https://github.com/tox-dev/tox/blob/7290a2ddad3728b94bfccdd358d028e9ba3ca5a1/src/tox/config/__init__.py#L245) is overriden.